core/base/CMakeLists.txt

```diff
@@ -198,11 +198,27 @@ set(BASE_SOURCES
   src/TVirtualX.cxx
 )
 
+set(RELATIVE_BASE_INC_HEADERS ${BASE_HEADERS})
+list(TRANSFORM RELATIVE_BASE_INC_HEADERS PREPEND inc/)
+
 # only here complete list of headers can be propogated to parent cmake file
 set_property(TARGET Core APPEND PROPERTY DICT_HEADERS ${BASE_HEADERS})
 
 target_sources(Core PRIVATE ${BASE_SOURCES})
 
+if(NOT CMAKE_VERSION VERSION_LESS "3.23.0") # https://discourse.cmake.org/t/file-set-xyz-is-listed-in-interface-file-sets-of-w-but-has-not-been-exported/9131/3
+  target_sources(
+      Core
+      PRIVATE
```

Comment:
@guitargeek thanks for the discussion. Followup: see this keyword here, one can easily separate private from public headers by doing:

```
target_sources(Core
  PUBLIC
  FILE_SET public_headersOrWhateverNameYouWant
  TYPE HEADERS
...


target_sources(Core
  PRIVATE
  FILE_SET private_headersOrWhateverNameYouWant
...
```

Right now, it might make more sense to first define everything is private headers without touching target_include_directories, ie we are just adding the headers for IDE purposes.

In a separate MR one could then remove all the target_include_directories by correctly defining which headers are public and which are private.